doc/cookbook_vs.md

```diff
@@ -373,6 +373,37 @@ The above code will force allocation of a thread for each degree of concurrency
 
 Note that being thread-safe is *not* the same thing as being free-threaded, which is an independent metric. Code can be free-threaded, thread-safe, both, or neither.
 
+## Should I await a Task with `.ConfigureAwait(false)`?
+
+### Short answer
+
+* Use `ConfigureAwait(false)` when writing app-independent library code. Such a library should avoid frequent use of `Task.Wait()` and `Task.Result`.
+* Use `ConfigureAwait(true)` when writing code where a `JoinableTaskFactory` is available. Use `await TaskScheduler.Default;` before CPU intensive tasks.
+
+**Note:** Use of `.ConfigureAwait(true)` is equivalent to awaiting a `Task` directly. Using this suffix is a way to suppress the warning emitted by some analyzers that like to see `.ConfigureAwait(false)` everywhere. Where no such analyzer is active, omitting the suffix is recommended for improved code readability.
+
+### Explanation
+
+Awaiting tasks with `.ConfigureAwait(false)` is a popular trend for a couple reasons:
+
+1. It allows the continuation (when scheduled) to run on the threadpool instead of returning to an unknown `SynchronizationContext` set by the caller. This can improve efficiency, and keep CPU intensive work off the UI thread, thereby increasing an application's responsiveness to user input.
+1. It can avoid deadlocks when people use `Task.Wait()` or `Task.Result` from the UI thread.
+
+But `.ConfigureAwait(false)` carries disadvantages as well:
+
+1. The tendency for `Task.Wait()` to work actually encourages such synchronously blocking code, yet deadlocks can still happen if the UI thread is actually required for one of the scheduled continuations.
+1. It makes for harder to read and maintain async code.
+1. It may not actually move CPU intensive work off the UI thread since it only makes the switch on the first *yielding* await.
+1. It contributes to [threadpool starvation][ThreadpoolStarvation] when the code using it is called in the context of a `JoinableTaskFactory.Run` delegate.
+
+The last disadvantage above deserves some elaboration. The `JoinableTaskFactory.Run` method sets a special `SynchronizationContext` when invoking its delegate so that async continuations automatically run on the original thread (without deadlocking). When awaits use `.ConfigureAwait(false)` it ignores the `SynchronizationContext` and defeats this optimization. As a result the scheduled continuation will execute on a (new) thread from the threadpool, even though the JTF.Run thread is blocked waiting for the delegate to complete and could have executed the continuation. In this scenario, *two* threads are allocated although only one is active.
```

Comment:
Is `.ConfigureAwait(false)` within a `JoinableTaskFactory.Run` _guaranteed_ to start a new thread? If so, why? I would have thought it would end up running on an existing threadpool thread--unless none were available within some time frame, and only then would a new thread be started up.

Reply:
I've reworded this to be more clear. I didn't mean to say it was always a *new* thread. I meant to say it was a thread from the threadpool, and could mean a new thread is created (in the case of threadpool starvation).